File: backend/api/src/main/kotlin/io/github/daegwonkim/backend/dto/user/GetUserLikedRentalItems.kt
package io.github.daegwonkim.backend.dto.user

import java.time.Instant

data class GetUserLikedRentalItemsResponse(
    val rentalItems: List<RentalItem>
) {
    data class RentalItem(
        val id: Long,
        val sellerId: Long,
        val title: String,
        val thumbnailImageUrl: String,
        val pricePerDay: Int,
        val pricePerWeek: Int,
        val address: String,
        val liked: Boolean,
        val stats: RentalItemStats,
        val createdAt: Instant,
    ) {
        data class RentalItemStats(
            val rentalCount: Int,
            val likeCount: Int,
            val viewCount: Int,
        )
    }
}

File: frontend/src/api/user/dto/GetUserLikedRentalItems.ts
export interface GetUserLikedRentalItemsResponse {
  rentalItems: RentalItem[]
}

interface RentalItem {
  id: number
  sellerId: number
  title: string
  thumbnailImageUrl?: string
  pricePerDay?: number
  pricePerWeek?: number
  address: string
  liked: boolean
  stats: RentalItemStats
  createdAt: Date
}

interface RentalItemStats {
  rentalCount: number
  likeCount: number
  viewCount: number
}
